panic: version string "v20220409-v1.23.0" doesn't match expected regular expression

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!-->


**What happened**:
Tests fail because `aws-cloud-controller-manager` fail to start:
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kops/13452/pull-kops-e2e-cni-cilium-ipv6/1516070708126420992
```
==== START logs for container aws-cloud-controller-manager of pod kube-system/aws-cloud-controller-manager-94ngm ====
panic: version string "v20220409-v1.23.0" doesn't match expected regular expression: "^v(\d+\.\d+\.\d+)"

goroutine 1 [running]:
k8s.io/component-base/metrics.parseVersion({{0x0, 0x0}, {0x0, 0x0}, {0x29703c0, 0x11}, {0x254765f, 0xb}, {0x0, 0x0}, ...})
	k8s.io/component-base@v0.23.5/metrics/version_parser.go:47 +0x27e
k8s.io/component-base/metrics.newKubeRegistry({{0x0, 0x0}, {0x0, 0x0}, {0x29703c0, 0x11}, {0x254765f, 0xb}, {0x0, 0x0}, ...})
	k8s.io/component-base@v0.23.5/metrics/registry.go:320 +0x118
k8s.io/component-base/metrics.NewKubeRegistry()
	k8s.io/component-base@v0.23.5/metrics/registry.go:335 +0x78
k8s.io/component-base/metrics/legacyregistry.init()
	k8s.io/component-base@v0.23.5/metrics/legacyregistry/registry.go:29 +0x1d
==== END logs for container aws-cloud-controller-manager of pod kube-system/aws-cloud-controller-manager-94ngm ====
```

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:
Try to start `aws-cloud-controller-manager` using the latest stable image: `registry.k8s.io/provider-aws/cloud-controller-manager:v1.23.0`.

**Anything else we need to know?**:
Guess it's related to: https://github.com/kubernetes/cloud-provider-aws/pull/317.

**Environment**:
- Kubernetes version (use `kubectl version`): v1.23.5
- Cloud provider or hardware configuration: AWS
- OS (e.g. from /etc/os-release): Ubuntu 22.04
- Kernel (e.g. `uname -a`):
- Install tools: kOps
- Others: n/a

<!-- DO NOT EDIT BELOW THIS LINE -->
/kind bug

--- a/cloudbuild.yaml
+++ b/cloudbuild.yaml
@@ -9,7 +9,7 @@ steps:
     - build
     - --tag=gcr.io/$PROJECT_ID/cloud-controller-manager:$_GIT_TAG
     - --tag=gcr.io/$PROJECT_ID/cloud-controller-manager:latest
-    - --build-arg=VERSION=$_GIT_TAG
+    - --build-arg=VERSION=$(echo $_GIT_TAG | awk -F - '{print $2}')
     - --output=type=registry
     - --platform=linux/amd64,linux/arm64
     - .

--- a/cloudbuild.yaml
+++ b/cloudbuild.yaml
@@ -9,7 +9,7 @@ steps:
     - build
     - --tag=gcr.io/$PROJECT_ID/cloud-controller-manager:$_GIT_TAG
     - --tag=gcr.io/$PROJECT_ID/cloud-controller-manager:latest
-    - --build-arg=VERSION=$(echo $_GIT_TAG | awk -F - '{print $2}')
+    - --build-arg=VERSION=$(echo $_GIT_TAG | cut -d - -f2-)
     - --output=type=registry
     - --platform=linux/amd64,linux/arm64
     - .

